aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -175,8 +175,15 @@ private void newStream() throws IOException {
       stream.close();
     }
 
+    boolean createStagingDirectory = false;
+    if (!stagingDirectory.exists()) {
+      createStagingDirectory = stagingDirectory.mkdirs();
+    }
     currentStagingFile = File.createTempFile("s3fileio-", ".tmp", stagingDirectory);
     currentStagingFile.deleteOnExit();
+    if (createStagingDirectory) {
+      stagingDirectory.deleteOnExit();
```

Comment:
why do we need to delete on exit? I think we can assume it's safe to keep the staging directory there.

Reply:
Yeah, it should be safe. Made the changes accordingly.